crates/dojo-lang/src/syntax/self_param.rs

```diff
@@ -0,0 +1,48 @@
+use cairo_lang_defs::plugin::PluginDiagnostic;
+use cairo_lang_diagnostics::Severity;
+use cairo_lang_syntax::node::db::SyntaxGroup;
+use cairo_lang_syntax::node::{ast, ids};
+
+use crate::syntax::utils as syntax_utils;
+
+const SELF_PARAM_NAME: &str = "self";
+
+/// Checks if the given function is not using `self` param.
+///
+/// # Arguments
+///
+/// - `db` - The syntax group.
+/// - `param_list` - The parameter list of the function.
+/// - `fn_diagnostic_item` - The diagnostic item of the function.
+/// - `diagnostics` - The diagnostics vector.
+pub fn check_self_parameter(
```

Comment:
```suggestion
/// Checks if the given function parameter is using `self` instead of `world` param. Reports error if thats the case.
///
/// # Arguments
///
/// - `db` - The syntax group.
/// - `param_list` - The parameter list of the function.
/// - `fn_diagnostic_item` - The diagnostic item of the function.
/// - `diagnostics` - The diagnostics vector.
pub fn check_parameter(
```

Reply:
Good point, thanks for the review!